src/Codec/CodecLibrary.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace MongoDB\Codec;
+
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Exception\UnsupportedValueException;
+
+class CodecLibrary implements Codec
+{
+    use DecodeIfSupported;
+    use EncodeIfSupported;
+
+    /** @var array<Decoder> */
+    private $decoders = [];
+
+    /** @var array<Encoder> */
+    private $encoders = [];
+
+    /** @param Decoder|Encoder $items */
+    public function __construct(...$items)
```

Comment:
Before we settle on this signature, I just want to confirm that you don't expect we'd ever need to solicit additional params in the constructor (e.g. options array).

If so, consider making this an array argument.

Reply:
I don't think so - this particular library is a first-come-first-serve library. If we have the need for different behaviour, I'd rather create a separate library class for the purpose.